lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -241,6 +278,47 @@ void MergeOp::getCanonicalizationPatterns(RewritePatternSet &results,
   results.insert<circt::handshake::EliminateSimpleMergesPattern>(context);
 }
 
+/// Returns a dematerialized version of the value 'v', defined as the source of
+/// the value before passing through a buffer or fork operation.
+static Value getDematerialized(Value v) {
```

Comment:
I'm not sure I follow where the "dematerialized" name comes from. Maybe use something more specific like "getUnbufferedValue"?

Reply:
Sounds good. Naming is hard. Since we "materialize" forks, buffers, etc., I am starting to understand the name "dematerialize".